cloudinit/util.py

```diff
@@ -2894,13 +2894,13 @@ def get_proc_ppid(pid):
     ppid = 0
     try:
         contents = load_file("/proc/%s/stat" % pid, quiet=True)
+        if contents:
+            # see proc.5 for format
+            m = re.search(r"^\d+ \(.+\) [RSDZTtWXxKWP] (\d+)", str(contents))
+            if m:
+                ppid = int(m.group(1))
```

Comment:
I think we probably should also WARN if we are unable to match here so we don't silently succeed returning 0 as parent process on unexpected format